CardsFiles/Cards.cpp

```diff
@@ -0,0 +1,251 @@
+        playingHand.handVector.pop_back(); ///< Remove the pointer of the card from the player's hand.
+    }
+}
+
+/**
+ * Default constructor of the Card class, creates a default card with "undefinedCardType"
+ * for its card type.
+ */
+Card::Card()
+{
+    cardType = "undefinedCardType";
+}
+
+/**
+ * Parametrized constructor of the Card class, creates a card with a given
+ * card type as its parameter.
+ * 
+ * @param cardType Given card type to the card instance.
+ */
+Card::Card(std::string cardType)
+{
+    this->cardType = cardType;
+}
+
+/**
+ * Overloaded Copy constructor of the Card class, creates a card given
+ * another instance of a card. This is a deep copy.
+ * 
+ * @param card Card instance that is being copied from.
+ */
+Card::Card(const Card& card)
+{
+    this->cardType = card.cardType;
+}
+
+/**
+ * Overloaded Assignment operator which assigns the card type of the assigning
+ * card instance to the assigned card instance.
+ * 
+ * @param card Card instance that is being assigned from.
+ */
+void Card::operator=(const Card& card)
+{
+    this->cardType = card.cardType;
+}
+
+/**
+ * Overloaded Stream insertion operator which outputs the card type 
+ * of the card instance.
+ * 
+ * @param COUT The output stream object.
+ * @param CARD The card instance that is being outputted.
+ */
+std::ostream& operator<<(std::ostream& COUT, const Card& CARD)
+{
+    COUT << "Card type: " << CARD.cardType << std::endl;
+    return COUT;
+}
+
+
+// <<<< Deck Class Definitions >>>> 
+
+
+/**
+ * Method that draws a card from the deck and places
+ * it in the hand of the given player.
+ * 
+ * @param hand The hand in which the drawn card will be placed.
+ */
+void Deck::draw(Hand& hand)
+{
+    if(this->deckVector.size() == 0)
+    {
+        std::cout << "Deck is empty, no card available to draw\n" << std::endl;
+    }
+    else
+    {
+        srand(time(0)); ///< Generate seed based on computer time for a random number when using rand().
+        int index = rand() % this->deckVector.size(); ///< Generate a random number between 0 and (DECK_VECTOR_SIZE - 1).
+        hand.handVector.push_back(this->deckVector[index]); ///< Append randomly chosen card to the player's hand.
+        this->deckVector.erase(this->deckVector.begin() + index); ///< Remove randomly chosen card from the deck.
+    }
+}
+
+/**
+ * Default constructor of the Deck class, creates a default deck with 
+ * no cards in it.
+ */    
+Deck::Deck()
+{
+    ///< Initializing empty deck
+}
+
+/**
+ * Overloaded Copy constructor of the Deck class, creates a deck given
+ * another instance of a deck. This is a deep copy.
+ * 
+ * @param copyPlayingDeck Deck instance that is being copied from.
+ */
+Deck::Deck(const Deck& copyPlayingDeck)
+{
+    for(int i = 0; i < copyPlayingDeck.deckVector.size(); i++)
+    {
+        this->deckVector.push_back(copyPlayingDeck.deckVector[i]); ///< Append all contents of passed deck to our deck.
```

Comment:
You should actually deference the cards and use it in the Card copy constructor in order to properly deep copy this vector. What is currently being done is that the pointers to the Cards in `copyPlayingDeck.deckVector` are being copied, not the actual card instances themselves.

ex: `this->deckVector.push_back(new Card(*copyPlayingDeck.deckVector[i]));`